packages/metal-quartz-collapse/src/MetalQuartzCollapse.js

```diff
@@ -0,0 +1,273 @@
+			this.collapsed ? this.close() : this.open();
+		}
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	disposeInternal() {
+		super.disposeInternal();
+		this.eventHandler_.removeAllListeners();
+		this.transitionEventHandler_.removeAllListeners();
+	}
+
+	/**
+	 * Animates close when `collapsed` is true
+	 */
+	animateClose() {
+		this.content.transitionType = 0;
+
+		this.content.setAttribute('style', 'height: ' + this.content.firstElementChild.offsetHeight + 'px;');
+		dom.addClasses(this.content, this.transitionClass);
+		dom.removeClasses(this.content, this.closedClass);
+		dom.removeClasses(this.content, this.openClass);
+
+		this.content.offsetHeight;
+		this.content.style.removeProperty('height');
+
+		this.shimUnsupportedTransition_(this.content);
+	}
+
+	/**
+	 * Animates open when `collapsed` is false
+	 */
+	animateOpen() {
+		this.content.transitionType = 1;
+
+		dom.removeClasses(this.content, this.closedClass);
+		dom.addClasses(this.content, this.transitionClass);
+		this.content.setAttribute('style', 'height: ' + this.content.firstElementChild.offsetHeight + 'px;');
+
+		this.shimUnsupportedTransition_(this.content);
+	}
+
+	/**
+	 * Adds CSS classes and properties to the `content` element when `collapsed`
+	 * is true
+	 */
+	close() {
+		dom.addClasses(this.content, this.closedClass);
+		dom.removeClasses(this.content, this.openClass);
+		dom.removeClasses(this.content, this.transitionClass);
+		this.content.setAttribute('aria-expanded', false);
+		this.content.style.removeProperty('height');
+	}
+
+	/**
+	 * Checks to see if browser supports CSS3 Transitions and returns the name 
+	 * of the transitionend event; returns false if it's not supported
+	 * @protected
+	 */
+	getTransitionEndEvent_() {
+		let el = document.createElement('metalQuartzTransitionEnd');
+
+		let transitionEndEvents = {
+			transition: 'transitionend',
+			WebkitTransition: 'webkitTransitionEnd',
+			MozTransition: 'transitionend',
+			OTransition: 'oTransitionEnd otransitionend'
+		};
+
+		for (let name in transitionEndEvents) {
+			if (el.style[name] !== undefined) {
+				return transitionEndEvents[name];
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Handles a `click` event on the headers.
+	 * @param {!Event} event
+	 * @protected
+	 */
+	handleClick_(event) {
+		this.toggle();
+	}
+
+	/**
+	 * Handles a `keydown` event on the headers.
+	 * @param {!Event} event
+	 * @protected
+	 */
+	handleKeydown_(event) {
+		if (event.keyCode === 13 || event.keyCode === 32) {
+			this.toggle();
+			event.preventDefault();
+		}
+	}
+
+	/**
+	 * Handles the `transitionend` event on the content.
+	 * @param {!Event} event
+	 * @protected
+	 */
+	handleTransitionEnd_(event) {
+		this.content.transitionType ? this.open() : this.close();
+
+		this.transitionEventHandler_.removeAllListeners();
+	}
+
+	/**
+	 * Adds CSS classes and properties to the `content` element when `collapsed`
+	 * is false
+	 */
+	open() {
+		dom.addClasses(this.content, this.closedClass);
```

Comment:
Just an update: I can add and remove more than one class with `dom.addClasses`, `dom.removeClasses`. I was using those with `dom.hasClass` previously and was running into issues. I rolled back the work around.